test/e2e/httpproxy/grpc_test.go

```diff
@@ -106,8 +104,7 @@ func testGRPCServicePlaintext(namespace string) {
 				grpc_retry.WithBackoff(grpc_retry.BackoffExponential(time.Millisecond * 10)),
 				grpc_retry.WithMax(20),
 			}
-			conn, err := grpc.DialContext(dialCtx, addr,
-				grpc.WithBlock(),
+			conn, err := grpc.NewClient(addr,
```

Comment:
Given the semantic changes from `DialContext` to `NewClient`, I _believe_ that  just dropping the "dial timeout" here is fine and should be covered by subsequent request timeouts; we'll see how the CI E2E runs looks